package/MDAnalysis/analysis/contacts.py

```diff
@@ -630,16 +628,11 @@ def run(self, store=True, force=False, start=0, stop=None, step=1, **kwargs):
                 self.qavg += self.q
                 if store:
                     records.append((frame, q1, n1))
-                out.write("{frame:4d}  {q1:8.6f} {n1:5d}\n".format(**vars()))
+                out.write("%(frame)4d  %(q1)8.6f %(n1)5d\n" % vars())
         if store:
             self.timeseries = np.array(records).T
-        n_frames = len(np.arange(
-            self.universe.trajectory.n_frames)[start:stop:step])
-        if n_frames > 0:
-            self.qavg /= n_frames
-        else:
-            logger.warn("No frames were analyzed. Check values of start, stop, step.") 
-            logger.debug("start={start} stop={stop} step={step}".format(**vars()))
+        n_frames = len(range(total_frames)[start_frame:end_frame:step_value])
```

Comment:
Nah, I meant the generator has overriden `__getitem__`, so doing `len` does your end-start / step thingy